Source/JavaScriptCore/wasm/WasmIRGeneratorHelpers.h

```diff
@@ -163,6 +163,37 @@ static inline void emitThrowImpl(CCallHelpers& jit, unsigned exceptionIndex)
     });
 }
 
+static inline void emitStructNewImpl(CCallHelpers& jit, unsigned typeIndex)
+{
+    ASSERT(Options::useWebAssemblyGC());
+
+    // Instance in argumentGPR0
+    // frame pointer in argumentGPR1
+    // arguments to the struct_new off of stack pointer
+
+    GPRReg scratch = GPRInfo::nonPreservedNonArgumentGPR0;
+    jit.loadPtr(CCallHelpers::Address(GPRInfo::argumentGPR0, Instance::offsetOfOwner()), scratch);
+    {
+        auto preciseAllocationCase = jit.branchTestPtr(CCallHelpers::NonZero, scratch, CCallHelpers::TrustedImm32(PreciseAllocation::halfAlignment));
```

Comment:
Should we make a materializeInstance helper? Why isn't it pinned, or why can't we rematerialize it to the pinned register here?

Reply:
Still confused about this